test/functional/api/buffer_spec.lua

```diff
@@ -35,8 +35,42 @@ describe('api/buf', function()
       -- There's always at least one line
       eq(1, curbuf_depr('line_count'))
     end)
-  end)
 
+    it('line_count has defined behaviour for unloaded buffers', function()
+      -- we'll need to know our bufnr for when it gets unloaded
+      local bufnr = request('nvim_buf_get_number', 0)
```

Comment:
`nvim_buf_get_number` is deprecated, `nvim_get_current_buf` is preferred for this, or just use `curbuf()` without args.